### PR TITLE
add SSH to macOS job

### LINKED ISSUE #2465: Support SSH for generic macOS job
Both, Linux

https://github.com/pytorch/test-infra/blob/9e8ad98a37ee2e6d3720744bbcb7e49e2f77a19a/.github/workflows/linux_job.yml#L104-L107

and Windows

https://github.com/pytorch/test-infra/blob/9e8ad98a37ee2e6d3720744bbcb7e49e2f77a19a/.github/workflows/windows_job.yml#L84-L87

already support it. Without this, debugging CI failures is just much harder.

### PR DESCRIPTION
**Pull request by pmeier**
Closes #2465.

### CODE REVIEW

**Comment by huydhn** [comment]
Argh, I accidentally start those test jobs just to see if this work.  But we need to be careful here.  AFAIK, MacOS M1 pet runners are persistent, so writing SSH key there will overwrite the existing one preventing maintainers to login to the runner.  We had a similar accident in the past.  Please at least gate the step with `runner.arch == 'x86_64'` to make sure that this is run only on MacOS x86_64

```
- name: Setup SSH
  if: runner.arch == 'x86_64'
  uses: ./test-infra/.github/actions/setup-ssh
  with:
  github-secret: ${{ github.token }}
```

File: .github/workflows/macos_job.yml [hunk 1]
```diff
@@ -77,6 +77,11 @@ jobs:
           ref: ${{ inputs.test-infra-ref }}
           path: test-infra
 
+      - name: Setup SSH
```
Comment (huydhn) [inline .github/workflows/macos_job.yml:80]:
```suggestion
      - name: Setup SSH
        if: runner.arch == 'x86_64'
```

**Comment by pmeier** [comment]
Thanks for the additional details. I guess that ties directly into #2086.

**Review by osalpekar — APPROVED** [review]

**Comment by pmeier** [comment]
Connection is timing out: https://github.com/pytorch/vision/actions/runs/4295997439/jobs/7487167420#step:5:17

**Comment by osalpekar** [comment]
@huydhn Do you have more context on the connection timeout issue?

**Comment by huydhn** [comment]
`169.254.169.254` is a magic IP that is used to retrieve ec2 information https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-data-retrieval.html.  It's only accessible within AWS.  I don't think setup-ssh with with GH runners at all.  macos-12 is the only runner we get from GitHub AFAIK.

For those runners, there are 2 options that I know:

* https://github.com/mxschmitt/action-tmate.  @DanilBaibak used this before.  It works but overwrite the SSH key, so it's important to restore the key after
* https://github.com/chkimes/actions-ssh.  GitHub folks refers me to this

**Comment by pmeier** [comment]
- @DanilBaibak is looking into https://github.com/mxschmitt/action-tmate
- https://github.com/chkimes/actions-ssh only works on Linux so not applicable here

Let's close this PR and have a go at it later.

**Pull request closed without merging**